src/types/select.ts

```diff
@@ -1,20 +1,272 @@
+  selected: [item: SelectItem<T>]
+  'add-item': []
+}
+
+/**
+ * @internal
+ */
+export interface SelectItemsSlots<T extends string | number> {
+  content(props: { item: SelectItem<T> }): any
+}
+
+export interface SelectFilterFunctionParams<T extends string | number> {
+  query: string
+  items: SelectItem<T>[]
 }
 
 export type SelectDropdownFooterTextPosition = 'sticky' | 'static'
+
+export interface SelectProps<T extends string | number, U extends boolean> {
+  /**
+   * To set the value of the select without using `v-model`.
+   * @default ''
+   */
+  modelValue?: (U extends true ? T | string : T) | '' | null
+
+  /**
+   * Attributes to be passed to underlying KPopover component.
+   * See KPopover's props for more info.
+   * @default {}
+   */
+  kpopAttributes?: Omit<PopoverAttributes, 'target' | 'trigger'>
+
+  /**
+   * Maximum height for dropdown container.
+   * @default '300px'
+   */
+  dropdownMaxHeight?: string
+
+  /**
+   * Label associated with the select element.
+   * @default ''
+   */
+  label?: string
+
+  /**
+   * Label attributes to be passed to underlying KLabel component.
+   * @default {}
+   */
+  labelAttributes?: LabelAttributes
+
+  /**
+   * The width of the select and popover's min-width.
+   * @default '100%'
+   */
+  width?: string
+
+  /**
+   * Placeholder to be displayed when no item is selected.
+   * @default ''
+   */
+  placeholder?: string
+
+  /**
+   * Items are JSON objects with required 'label' and 'value'.
+   * {
+   *   label: 'Item 1',
+   *   value: 'item1'
+   * }
+   * @default []
+   */
+  items?: U extends true ? SelectItem<T | string>[] : SelectItem<T>[]
+
+  /**
+   * Control whether the select supports filtering.
+   * @default false
+   */
+  enableFiltering?: boolean
+
+  /**
+   * Override default filter functionality of case-insensitive search on label.
+   * @default (params: SelectFilterFunctionParams) => params?.items?.filter((item: SelectItem) => item.label?.toLowerCase().includes(params.query?.toLowerCase()))
+   */
+  filterFunction?: (params: SelectFilterFunctionParams<U extends true ? T | string : T>) =>
+    | U extends true
+      ? SelectItem<T | string>[]
+      : SelectItem<T>[]
+    | true
+
+  /**
+   * Loading state in autosuggest.
+   * @default false
+   */
+  loading?: boolean
+
+  /**
+   * A flag for clearing selection.
+   * @default false
+   */
+  clearable?: boolean
+
+  /**
+   * Dropdown footer text.
+   * @default ''
+   */
+  dropdownFooterText?: string
+
+  /**
+   * Dropdown footer text position.
+   * Accepted values: 'sticky' and 'static'.
+   * @default 'sticky'
+   */
+  dropdownFooterTextPosition?: SelectDropdownFooterTextPosition
+
+  /**
+   * If true and item-template is passed, will display item-template content inside selected-slot-template.
+   * @default false
+   */
+  reuseItemTemplate?: boolean
+
+  /**
+   * Allow creating new items.
+   * @default false
+   */
+  enableItemCreation?: boolean & U
```

Comment:
Using this (`boolean & U`) instead of `U` alone is a necessary workaround to ensure the Vue compiler treats `enableItemCreation` as a boolean prop at runtime.